internal/kubernetes/operator/deployment/deployment_test.go

```diff
@@ -520,3 +520,51 @@ func TestBuildServerlessDeployments(t *testing.T) {
 		}
 	})
 }
+
+func TestCleanTenantFields(t *testing.T) {
+	spec := akov2.AtlasDeploymentSpec{
+		DeploymentSpec: &akov2.AdvancedDeploymentSpec{
+			BackupEnabled: pointer.Get(true),
+			BiConnector: &akov2.BiConnectorSpec{
+				Enabled:        pointer.Get(true),
+				ReadPreference: "SECONDARY",
+			},
+			EncryptionAtRestProvider: "AWS",
+			PitEnabled:               pointer.Get(true),
+			MongoDBMajorVersion:      "5.1",
+			ReplicationSpecs: []*akov2.AdvancedReplicationSpec{
+				{
+					RegionConfigs: []*akov2.AdvancedRegionConfig{
+						{
+							ProviderName: "TENANT",
+						},
+					},
+				},
+			},
+		},
+	}
```

Comment:
make this a table test with a couple of cases:
1. DeploymentSpec is nil
2. ReplicationSpecs is nil
3. RegionConfigs is nil
4. RegionConfigs has multiple entries which does not include any "TENANT"
5. RegionConfigs has multiple entries which does include "TENANT"

Reply:
done!